plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -1330,6 +1360,85 @@ private static <K, V> Cache<K, AtomicReference<V>> buildBulkCache(
         return cacheBuilder.build();
     }
 
+    private class TableNamesCache
+    {
+        private static final String DUMMY_CACHE_KEY = "$KEY";
```

Comment:
You can do like https://github.com/trinodb/trino/blob/70189fc3bd9b706bcb4453893c27d930e6c631ed/plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerConnectionFactory.java#L110 to indicate the dummy cache key is never mixed with real schema names.